File: aml-frontend-private/src/model/HeaderBlockProps.ts
import type { BlockProps } from "./BlockProps";

export type HeaderLevel = "H1" | "H2" | "H3";
export type HeaderTextAlignment = "LEFT" | "CENTER" | "RIGHT";

export interface HeaderBlockProps extends BlockProps {
  text: string;
  level: HeaderLevel;
  alignment: HeaderTextAlignment;
}

File: aml-backend/src/main/java/dev/abarmin/aml/dashboard/block/header/HeaderBlockProps.java
package dev.abarmin.aml.dashboard.block.header;

import dev.abarmin.aml.dashboard.domain.BlockProps;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class HeaderBlockProps implements BlockProps {
  @NotEmpty
  private String text;

  @NotNull
  private HeaderLevel level;

  @NotNull
  private TextAlignment alignment;
}
